Src/IronPython/Modules/Builtin.cs

```diff
@@ -1037,27 +1098,44 @@ public class mapobject : IEnumerable {
                 }
                 return args[retIndex];
             } else {
-                throw PythonOps.TypeError("min expecting 1 arguments, got 0");
+                throw PythonOps.TypeError("min expected 1 arguments, got 0");
             }
         }
 
-        private static object GetMinKwArg([ParamDictionary]IDictionary<object, object> dict) {
-            if (dict.Count != 1)
-                throw PythonOps.TypeError(" min() should have only 1 keyword argument, but got {0} keyword arguments", dict.Count);
+        private static Tuple<object,object> GetMinKwArg([ParamDictionary]IDictionary<object, object> dict, bool isDefaultAllowed) {
+            if (dict.Count != 1 && dict.Count != 2)
+                throw PythonOps.TypeError(" min() should have only 2 keyword arguments, but got {0} keyword arguments", dict.Count);
 
+            if (dict.Keys.Contains("default") && !isDefaultAllowed)
+                throw PythonOps.TypeError(" Cannot specify a default for min() with multiple positional arguments");
+            
             return VerifyKeys("min", dict);
         }
 
-        private static object VerifyKeys(string name, IDictionary<object, object> dict) {
-            object value;
-            if (!dict.TryGetValue("key", out value)) {
-                ICollection<object> keys = dict.Keys;
-                IEnumerator<object> en = keys.GetEnumerator();
-                if (en.MoveNext()) {
-                    throw PythonOps.TypeError(" {1}() got an unexpected keyword argument ({0})", en.Current, name);
-                }
+        private static Tuple<object, object> VerifyKeys(string name, IDictionary<object, object> dict) {
+            object method;
+            object def;
+
+            var MyCopiedDictionary = new Dictionary<object, object>(dict);
```

Comment:
Would be nice to avoid the extra dictionary allocation here. You can probably do something like:
```
int cnt = 0;
if (dict.TryGetValue(...)) {
    cnt++;
}

if (dict.Count > cnt) {
    throw ...
}
```